consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -297,10 +299,17 @@ public void updateForkChoice(final Hash headBlockHash, final Hash finalizedBlock
           blockchain.setFinalized(blockHeader.getHash());
           mergeContext.setFinalized(blockHeader);
         });
+
+    return ForkchoiceResult.withResult(newFinalized, Optional.of(newHead));
   }
 
   @Override
   public boolean latestValidAncestorDescendsFromTerminal(final BlockHeader blockHeader) {
+    if (blockHeader.getNumber() <= 1L) {
```

Comment:
i do not trust this, but i believe it. I can't think of a counter to this assumption.

Reply:
Yeah, blockNumber could be negative which would be hugely pathological.  We could make this `== 0` instead

Reply:
whoops scratch that - I forgot this condition is checking for blocknumber in [0,1]

Reply:
I added a bit of safety to ensure we are past TTD.